language-design.md

```diff
@@ -28,28 +102,114 @@ you know a poorly designed language? What are the symptoms?
+better. Here the flexibility has a trade-off with consistency of expression.
+
+Anthony slightly disagrees with the statement on should.js. He really likes
+the idea of a language where you can use optional statements to make it
+more readable without the "syntactic aspartame" being necessary.
+
+Well-designed languages should have errors in mind: not every program is going
+to be perfect, so there needs to be an elegant way of handling errors. It should
+not be an afterthought. However, this verges on prgamming language design versus
+compiler design.
 
 ---
 
 **Question**
-
 How do the themes of _Growing a Language_ relate to the "sound lab" we did this week?
 
 **Response**
 
-
+In _Growing a Language_, there was an emphais on "patterns": creating a language
+and use pattern is essential to usability, learnability, and extensibility. This
+relates to the other theme of fluency. This was especially exemplified by the
+entire talk defining a language made up of single syllable words. Steele
+describes that a language should be a mode of discussion between a user and the
+bytes that language creates.
+
+We felt that these themes were the main themes of the sound lab as well. One
+goal of the lab was to allow for a better chaining of commands: an obvious way
+to increase the volume and reverse the sound helps to design a use pattern
+for the language. Also, the old sound code required the addition of `out.wav`
+for all function calls but the first one, which does not follow an overall
+pattern&mdash;or at the very best creates an unfriendly pattern. Bad patterns
+are worse than no patterns.
+
+In the "sound lab", as partners attempted to redesign the language in a very
+rushed amount of time, it felt as if we were coming up with a "master plan", or
+in Steele's words a "Cathedral-style" design.
+
+_Growing a Language_ also discussed the ability to extend a language which may
+or may not produce natural-feeling results depending on the original language
+design. When completing the "sound lab," groups seeemed to be thinking about
+adding new types of functions/sound manipulations and how that would feel or
+look in the API.
 
 ---
- 
-**Question**
 
+**Question**
 
-In what way is an API a language? 
+In what way is an API a language?
 
 **Response**
 
-
+Application Programming Intefaces are fundamentally subsets of a language. APIs
+serve to act as a (limited) system of communication&mdash;the definition of a
+language itself. It's not strictly about the end result either&mdash;there needs
+be a sense of flucency and "APIs should be self documenting" and "should rarely
+require documentation to read code written in a good API" [Bloch, 2006]. If you
+think about natural language, this is also true: once you have a basic
+understanding of a language, you rarely need to look words up in a dictionary
+since context (most of the time) can give you enough information to figure out the
```

Comment:
I agree that while being limited in scope, API's should definitely have a sense of fluency; in this way, I think designing an API can be very similar to designing a DSL. I really like the natural language analogy you give - that a basic understanding of an API should be enough for us to communicate what we want to do with it.